src/Illuminate/Database/Eloquent/Relations/BelongsTo.php

```diff
@@ -75,7 +75,10 @@ public function __construct(Builder $query, Model $child, $foreignKey, $ownerKey
      */
     public function getResults()
     {
-        return $this->query->first() ?: $this->getDefaultFor($this->parent);
+        return ! is_null($this->child->{$this->foreignKey})
+                    && ($results = $this->query->first())
+                ? $results
+                : $this->getDefaultFor($this->parent);
```

Comment:
FTR, I don't really like this style. It's really hard to miss and quickly understand the code because of the "ANDed operation with assign":
![image](https://user-images.githubusercontent.com/87493/50539871-ec08c900-0b87-11e9-90ff-57a94d1238a5.png)

Reply:
Yeah, it's not ideal. I'll change it.

Reply:
Nice change, thanks 👍